File: sdkwork-ai-sdk-go/api/rerank2.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type Rerank2Api struct {
    client *sdkhttp.Client
}

func NewRerank2Api(client *sdkhttp.Client) *Rerank2Api {
    return &Rerank2Api{client: client}
}

// Rerank documents
func (a *Rerank2Api) Rerank(body sdktypes.RerankRequest) (sdktypes.RerankResponse, error) {
    raw, err := a.client.Post(AiApiPath("/v1/rerank"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.RerankResponse
        return zero, err
    }
    return decodeResult[sdktypes.RerankResponse](raw)
}

File: sdkwork-ai-sdk-flutter/lib/src/api/rerank2.dart
import '../http/client.dart';
import '../models.dart';

class Rerank2Api {
  final HttpClient _client;
  
  Rerank2Api(this._client);

  /// Rerank documents
  Future<RerankResponse?> rerank(RerankRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/rerank'), body: body, contentType: 'application/json');
    return response is RerankResponse ? response : null;
  }
}
